### PR TITLE
[14.0][MIG] stock_reception_screen_mrp_subcontracting

### PR DESCRIPTION
**Pull request by sebalix**

### CODE REVIEW

File: stock_reception_screen_mrp_subcontracting/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        # context key and we rewrite the business logic.
+        new_self = self
+        if any(picking._is_subcontract() for picking in self):
+            new_self = self.with_context(subcontracting_skip_action_done=True)
+        res = super(StockPicking, new_self)._action_done()
+        for picking in self.filtered(lambda p: p._is_subcontract()):
+            productions = picking.move_lines.move_orig_ids.production_id.filtered(
+                lambda p: p.state not in ("done", "cancel")
+            )[-1:]
+            for production in productions:
```
Comment (mmequignon) [inline stock_reception_screen_mrp_subcontracting/models/stock_picking.py:30]: you should only have the last on of the recordset, right? If so, we don't need the `for` loop. Or am I missing smth?

**Review by mmequignon — APPROVED** [review]
Didn't get everything. Code looks good

File: stock_reception_screen_mrp_subcontracting/models/stock_picking.py (hunk 1 above)
Comment (sebalix) [reply stock_reception_screen_mrp_subcontracting/models/stock_picking.py:30]: Indeed, but I don't know if I should remove the for loop or the `[-1:]` (copied from Odoo std code), Odoo is acting as only one MO could be linked to a receipt, but I don't know what is blocking to have a receipt with 2 differents subcontracted product, with each of them having its MO... for this case we would need to loop.

To check.

**Comment by sebalix** [comment]
Build is failing because of a conflict between `shopfloor` and `mrp_subcontracting` (reproduced locally with only these two addons installed), not related to this addon especially. Checking...

**Review by i-vyshnevska — APPROVED** [review]

**Comment by simahawk** [comment]
@sebalix any news?

**Comment by sebalix** [comment]
This PR is working fine on production so I'll merge it. But if we could get rid of this wobbly screen for further versions (and switch to something based on Shopfloor) I'm OK with that.

**Comment by sebalix** [comment]
Something is broken... to check later

**Comment by flachica** [comment]
Hello @sebalix. I can't see the migrations issue to 15.0. I'm working to migrate shopfloor to 15.0. Where is the correct Issue to say this?

**Pull request closed without merging**